### DISCUSSION TITLE
Provisional Snatch with operator approval

### DISCUSSION THREAD

**Discussion by mmhere** [pymedusa/Medusa#10927] [Ideas & Feature requests]
### Suggested Enhancement

Have an option to designate a show (or particular providers, especially torrent providers) as **Provisional** and requiring operator review before an episode is considered Snatched with downloading underway.

**When a provisional snatch occurs**

Notify operator that manual interaction is suggested before proceeding.  Leave the show/episode in Wanted state so that background searches for other releases continue.

Place downloaded .torrent or .nzb in either a "proposed" directory for review, or dump into downloader's inbox to begin processing, with the understanding that it may be superseded by better releases chosen manually.  Perhaps such a provisional release could even be superseded automatically if a clear winner appears via other providers in the near future.

Sometimes you just don't know until something else appears that a better item is available -- you want to interact to make this choice but be notified that activity is occurring.

### Tl;DR

Bear with me:  This is a little bit tricky to explain.  (Maybe I'll edit this down but I wanted to put it out there.)

Torrents often appear before other sources but depending upon the tracker community in use and the behaviour of its peers, may be less reliable (or at least slower) than usenet providers.

For many releases, the item available via torrent soon becomes available via usenet.  When this happens the usenet feed is often preferred because completion, once snatched is quicker and more reliable.

Torrents are sometimes the only source.  When this is so, you want the torrent to be snatched and considered fully "in progress" by grabbing it and automatically delivering it to the downloader.

But in cases where, based on history and manual observation over time, you know that a torrent will soon become available via usenet, you'd like to prefer the .nzb over the .torrent.  So it would be nice to have a way to designate certain shows (or even torrent providers?) as *provisional* sources.

That is, when a torrent is snatched, it would be nice to download the torrent, get a notification, but leave the show in the Wanted state so that searching will continue on the usenet side (or even for other torrent providers, given their reliability varies).

You'd still like to choose as events transpire, capper patterns vary over time, and peer performance (for torrents) improves or degrades.

Perhaps the snatched episode could be placed in an "operator approval requested" state that references the snatched file.  Meaning, for torrents, the .torrent is grabbed, the operator is notified so they can change the state to "yes I'm letting the grabber get this", while regular background searching for other sources continues.

Basically, I'd like a way to default snatches (especially torrents) to a provisional-still-searching state.  The background search process would continue, the snatched file (.torrent or .nzb) would deposited somewhere (perhaps in a provisional black hole location), and the operator would be notified to review and "do something".

Although not intended for .nzb's in the above scenario, treating them this way could also be useful.  I find that in my setup .nzb's are more reliable (more likely to complete and do so quickly) and that usually the best one is chosen automatically.  Provisional status could, however, also prove useful here.

Sometimes you have a show set to a mixture of preferred Qualities and you want to manually choose one when several appear, or defer the first one that appears because *usually* a better/preferred one appears soon after.  This may vary for a show over time as cappers get better or worse at capping.  Your experience as an operator (which one to allow to download) also improves over time.

Here, and especially for shows that have a wide variety of releases qualities available, you might like to review that list before the automation chooses one.  If you can predict that the first one to appear will soon be supplanted by a preferred quality, you might choose to skip the first release-quality that appears because you're pretty sure a preferred one will appear soon after.

I know, carefully setting the Quality preferences for such a show usually automates all of this;  but sometimes you have a better prediction for what will occur based on experience and recent behaviour in the marketplace as that behaviour varies over time.
